packages/unchained-client/src/ethereum/parser/uniV2.ts

```diff
@@ -131,6 +142,35 @@ export class Parser implements SubParser {
     }
   }
 
+  async parseStakingRewards(tx: BlockbookTx): Promise<TxSpecific | undefined> {
+    const txData = tx.ethereumSpecific?.data
+
+    if (!txData) return
+
+    const txSigHash = getSigHash(txData)
+
+    if (!Object.values(this.supportedStakingRewardsFunctions).some((hash) => hash === txSigHash))
+      return
+
+    const decoded = this.stakingRewardsInterface.parseTransaction({ data: txData })
+
+    // failed to decode input data
+    if (!decoded) return
+
+    return {
+      data: {
+        parser: TxParser.UniV2,
+        method: decoded.name
+      }
+    }
+  }
+
+  async parse(tx: BlockbookTx): Promise<TxSpecific | undefined> {
+    if (txInteractsWithContract(tx, UNI_V2_ROUTER_CONTRACT)) return this.parseUniV2(tx)
+    if (txInteractsWithContract(tx, UNI_V2_STAKING_REWARDS)) return this.parseStakingRewards(tx)
```

Comment:
This will work for parsing stake/exit against the `0x212ebf9FD3c10F371557b08E993eAaB385c3932b` fox staking rewards contract, but will not catch any other staking rewards. Since there is ultimately a huge list of contracts that can support this functionality we probably want to parse any transaction that has input data that is able to be decoded using the `stakingRewardsInterface`. This does raise the previous question about ideally wanting to have generated asset data for all uniV2 token pairs

Reply:
Agree, though that's something we should either do as a separate bounty or an internal piece of work.

@kaladinlight are you happy with a descriptive TODO comment here in the interim?

Reply:
@0xApotheosis yes, that works fine for now